pkg/antctl/command_definition_test.go

```diff
@@ -439,3 +440,612 @@ func TestCommandDefinitionGenerateExample(t *testing.T) {
+								shorthand:    "S",
+							},
+							{
+								name:         "networkpolicy",
+								defaultValue: "NetworkPolicy",
+								usage:        "NetworkPolicy name. If present, Namespace must be provided.",
+								shorthand:    "N",
+							},
+							{
+								name:         "table",
+								defaultValue: "Table",
+								usage:        "Comma separated Antrea OVS flow table names or numbers",
+								shorthand:    "T",
+							},
+							{
+								name:         "groups",
+								defaultValue: "Groups",
+								usage:        "Comma separated OVS group IDs. Use 'all' to dump all groups",
+								shorthand:    "G",
+							},
+						},
+					},
+				},
+				commandGroup: get,
+			},
+			expected: map[string]string{"groups": "Groups", "namespace": "test1", "networkpolicy": "NetworkPolicy", "pod": "Pod", "service": "Service", "table": "Table"},
+			mode:     "agent",
+			args:     []string{"test1", "test2"},
+		},
+		{
+			name: "Command for flowaggregator defines resource endpoint",
+			cd: &commandDefinition{
+				use: "podmuticasts",
+				flowAggregatorEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+				controllerEndpoint: &endpoint{},
+				agentEndpoint:      &endpoint{},
+			},
+			mode:     "flowaggregator",
+			expected: map[string]string{"name": "test1", "namespace": ""},
+			args:     []string{"test1", "test2"},
+		},
+		{
+			name: "Command for controller defines non-resource endpoint",
+			mode: "controller",
+			cd: &commandDefinition{
+				use:          "endpoint",
+				aliases:      []string{"endpoints"},
+				short:        "Filter network policies relevant to an endpoint.",
+				long:         "Filter network policies relevant to an endpoint into three categories: network policies which apply to the endpoint and policies which select the endpoint in an ingress and/or egress rule.",
+				commandGroup: query,
+				controllerEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/endpoint",
+						params: []flagInfo{
+							{
+								name:            "namespace",
+								defaultValue:    "default",
+								supportedValues: []string{"default", "ns1", "ns2"},
+								usage:           "Namespace of the endpoint (defaults to 'default')",
+								shorthand:       "n",
+							},
+							{
+								name:            "pod",
+								defaultValue:    "Pod",
+								supportedValues: []string{"pod1", "pod2", "pod3"},
+								usage:           "Name of a Pod endpoint",
+								shorthand:       "p",
+							},
+						},
+					},
+				},
+			},
+			expected:      map[string]string(nil),
+			expectedError: "unsupported value Pod for flag pod",
+			args:          []string{"test1", "test2"},
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			cmd := &cobra.Command{
+				Use:     tt.cd.use,
+				Aliases: tt.cd.aliases,
+				Short:   tt.cd.short,
+				Long:    tt.cd.long,
```

Comment:
Do we need all these parameters? I didn't see you set arg in any of `[]flagInfo{}`, I feel you can keep minimal test data if they are not required for tests. Please double check.

Reply:
removed unnecessary parameters